shell/platform/android/io/flutter/embedding/android/AndroidTouchProcessor.java

```diff
@@ -220,6 +224,28 @@ private void addPointerForIndex(
     }
 
     int pointerKind = getPointerDeviceTypeForToolType(event.getToolType(pointerIndex));
+    // We use this in lieu of using event.getRawX and event.getRawY as we wish to support
+    // earlier versions than API level 29.
+    float viewToScreenCoords[] = {event.getX(pointerIndex), event.getY(pointerIndex)};
+    transformMatrix.mapPoints(viewToScreenCoords);
+    long buttons;
+    if (pointerKind == PointerDeviceKind.MOUSE) {
+      buttons = event.getButtonState() & 0x1F;
+      if (buttons == 0
+          && event.getSource() == InputDevice.SOURCE_MOUSE
+          && pointerChange == PointerChange.DOWN) {
+        // Some implementations translate trackpad scrolling into a mouse down-move-up event
+        // sequence with buttons: 0, such as ARC on a Chromebook. See #11420, a legacy
+        // implementation that uses the same condition but converts differently.
+        ongoingPans.put(event.getPointerId(pointerIndex), viewToScreenCoords);
+      }
```

Comment:
```suggestion
    if (pointerKind == PointerDeviceKind.MOUSE) {
      buttons = event.getButtonState() & 0x1F;
    } else if (pointerKind == PointerDeviceKind.TOUCH) {
      buttons = 0;
      if (event.getButtonState() == 0
          && event.getSource() == InputDevice.SOURCE_MOUSE
          && pointerChange == PointerChange.DOWN) {
        // Chromebook with ARC translates trackpad scrolling into a touch down-move-up event
        // sequence, which can be identified by a distinctive combination of input device "mouse",
        // tool type "finger", and buttons 0.
        ongoingPans.put(event.getPointerId(pointerIndex), viewToScreenCoords);
      }
```

Reply:
Can you also add a test case for this special case?

Reply:
Any concern that we have no idea which proportion of devices this will work on? I guess it's based on undocumented behaviour anyway, and Chrome OS is always up to date.

Reply:
Also maybe it's better to use `switch` instead.

Reply:
Never mind, I'll handle this change in another PR. The current code passes all tests anyway.